src/phoenix/core/model_schema_adapter.py

```diff
@@ -137,3 +150,37 @@ def _translate_embedding(
         link_to_data=embedding.link_to_data_column_name,
         display_name=display_name,
     )
+
+
+def _guess_vectors_vs_scalars(
+    names: Iterable[str],
+    *dataframes: pd.DataFrame,
+) -> Tuple[List[str], List[Embedding]]:
+    """A best-effort attempt at separating vector columns from scalar columns
+    by examining the first non-null item of the column from each dataframe. If
+    any item is `Iterable` and `Sized`, but not `str`, then the column is
+    returned as `Embedding`, else it's returned as scalar.
+    """
+    scalars: List[str] = []
+    vectors: List[Embedding] = []
+    # convert to sets for a speedier lookup
+    column_names = [set(df.columns) for df in dataframes]
+    for name in names:
+        for i, df in enumerate(dataframes):
```

Comment:
maybe I'm just not familiar but these are really the "columns" not `df` right? naming has got me confused.

Reply:
these are multiple (and variadic) dataframes, i.e. including primary and reference